### PR TITLE
Feature: WhatsApp structured messages documentation. RD-18844

### PR DESCRIPTION
**Pull request by alexeevit**
Here's the documentation for the WhatsApp brand new List Messages and Button Replies structured messages. I took all screenshots by myself so review them too pls.

### CODE REVIEW

File: docs/interactions/structured-messages/index.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Engage Digital provides an omnichannel API to create structured messages. Since
   | **Google Business Messages** | **<span style="color:green">Yes</span>** | **<span style="color:green">Yes</span>** | **<span style="color:green">Yes</span>** | **<span style="color:green">Yes</span>** | N/A | N/A |  N/A |
   | **Source SDK** | **<span style="color:green">Yes</span>** | **<span style="color:green">Yes</span>** | **<span style="color:green">Yes</span>** | **<span style="color:green">Yes</span>** | N/A | N/A |  N/A |
   | **Twitter Direct Messages** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | N/A | N/A | N/A |  N/A |
-  | **WhatsApp** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | **<span style="color:orange">Not yet</span>** | N/A | N/A | N/A |  N/A |
+  | **WhatsApp** | **<span style="color:green">Yes</span>** | **<span style="color:orange">Not yet</span>** | **<span style="color:green">Yes</span>** | N/A | N/A | N/A |  N/A |
```
Comment (CedricBm) [inline docs/interactions/structured-messages/index.md:42]: I think you can also change the rich link support to N/A

File: docs/interactions/structured-messages/template.md [hunk 2]
```diff
@@ -113,3 +113,30 @@ Primary parameters are used by default, however, some parameters are unique or o
...
+
+Primary parameters are used by default, however, some parameters are unique or overwritten by parameters specific to this example.
+
+| API Property | Type | Specificity |
+|-|-|-|
+| **If subtitle is present** | | |
+| **`structured_content.title`** | String | Truncated to 60 characters. |
+| **`structured_content.subtitle`** | String | Truncated to 963 characters if an attachment is present. |
+|
+| **If subtitle is empty** | | |
```
Comment (CedricBm) [inline docs/interactions/structured-messages/template.md:131]: Looks like this doesn't render correctly, on markdown at least. Maybe I can check the end result somewhere?

Screenshot from 2021-09-08 15-12-53

File: docs/interactions/structured-messages/template.md [hunk 3]
```diff
@@ -113,3 +113,30 @@ Primary parameters are used by default, however, some parameters are unique or o
...
+
+## Example: WhatsApp (Reply Buttons)
+
+<img class="img-fluid" width="398" src="../../../img/structured-messages-template-whatsapp.png">
+
+### Properties Unique to this Channel
+
+Primary parameters are used by default, however, some parameters are unique or overwritten by parameters specific to this example.
+
+| API Property | Type | Specificity |
```
Comment (CedricBm) [inline docs/interactions/structured-messages/template.md:125]: You should also mention somewhere that url isn't supported on both the body and the items. (Yes I know you already put "reply only" in the item, but better be real clear)

File: docs/interactions/structured-messages/select.md [hunk 4]
```diff
@@ -220,3 +220,78 @@ Primary parameters are used by default, however, some parameters are unique or o
...
+### JSON Body
+
+```json
+{
+  "source_id": "<source_id>",
+  "in_reply_to_id": "<in_reply_to_id>",
+  "body": "What do you wish?",
+  "structured_content": {
+    "type": "select",
+    "button": "Choose an option",
```
Comment (CedricBm) [inline docs/interactions/structured-messages/select.md:240]: Can you add all the specific attributes in this example payload? Ideally the screenshots should be the result of this payload.

File: docs/interactions/structured-messages/select.md [hunk 5]
```diff
@@ -220,3 +220,78 @@ Primary parameters are used by default, however, some parameters are unique or o
...
+
+| API Property | Type | Description |
+|-|-|-|
+| **Structured Content Settings** | | |
+| **`structured_content.items`** | Array | Truncated to 10 elements. |
+| **`structured_content.button`** | String | **Optional**. The button text field.<br>Limited to 20 characters. "See options" by default. |
+| **`structured_content.title`** | String | **Optional**. The title text field.<br>Limited to 60 characters.  |
+| **`structured_content.footer`** | String | **Optional**. The footer text field.<br>Limited to 60 characters.  |
+|
+| **Section Settings** | | |
```
Comment (CedricBm) [inline docs/interactions/structured-messages/select.md:288]: Same comment about rendering on markdown here:

Screenshot from 2021-09-08 15-19-20

**Review by CedricBm — CHANGES_REQUESTED** [review]

File: docs/interactions/structured-messages/select.md (hunk 5 above)
Comment (alexeevit) [reply docs/interactions/structured-messages/select.md:288]: fixed

File: docs/interactions/structured-messages/select.md (hunk 4 above)
Comment (alexeevit) [reply docs/interactions/structured-messages/select.md]: fixed

File: docs/interactions/structured-messages/template.md (hunk 3 above)
Comment (alexeevit) [reply docs/interactions/structured-messages/template.md:125]: yep, added

File: docs/interactions/structured-messages/template.md (hunk 2 above)
Comment (alexeevit) [reply docs/interactions/structured-messages/template.md:130]: fixed

File: docs/interactions/structured-messages/index.md (hunk 1 above)
Comment (alexeevit) [reply docs/interactions/structured-messages/index.md]: changed, I also sorted it alphabetically

File: docs/interactions/structured-messages/select.md [hunk 6]
```diff
@@ -284,14 +285,11 @@ Primary parameters are used by default, however, some parameters are unique or o
...
 | **`structured_content.footer`** | String | **Optional**. The footer text field.<br>Limited to 60 characters.  |
-|
 | **Section Settings** | | |
 | **`structured_content.sections`** | Array | **Optional**. Limited to 10 elements. |
 | **`structured_content.sections.title`** | String | **Optional if there's only a single section**. The title of the section.<br>Limited to 24 characters. |
 | **`structured_content.sections.identifier`** | String | Identifier of the section that will be used to organize items in the section.<br>Limited to 200 characters. |
-|
 | **Item Settings** | | |
 | **`structured_content.items.title`** | String | The item title field.<br>Truncated to 24 characters. |
-| **`structured_content.items.payload`** | String | **Optional**. The item payload field.<br>Limited to 200 characters.<br>Automatically gets populated as a random hex if blank. |
```
Comment (CedricBm) [inline docs/interactions/structured-messages/select.md]: We still need this one because the limit is different from the generic payload.

Comment (CedricBm) [reply docs/interactions/structured-messages/select.md]: And you can probably say it's autopopulated too.

**Comment by CedricBm** [comment]
One last comment then LGTM

File: docs/interactions/structured-messages/select.md (hunk 6 above)
Comment (alexeevit) [reply docs/interactions/structured-messages/select.md]: ah, I removed it accidentally, fixed

**Comment by alexeevit** [comment]
> One last comment then LGTM

@CedricBm since I'm on vacation, feel free to merge it :)

**Comment by CedricBm** [comment]
Thanks a lot for finishing this. I'll handle the merge tomorrow.

**Pull request merged**